src/components/ScaleLattice.vue

```diff
@@ -0,0 +1,338 @@
+});
+
+onUnmounted(() => {
+  window.removeEventListener("resize", adjustLatticeSize);
+});
+
+// === Computed state ===
+
+const canLatticePrimes = computed(() => {
+  try {
+    return props.scale.intervals.reduce(
+      (canLattice: boolean, interval: Interval) =>
+        canLattice &&
+        interval.monzo
+          .toIntegerMonzo()
+          .slice(9) // hardcoded for now to the 23-limit
+          .reduce(
+            (isZero: boolean, component: number) => isZero && component === 0,
+            true
+          ),
+      true
+    );
+  } catch {
+    return false;
+  }
+});
+
+function maybeGetPrimeEquave(equave: Interval): number | null {
+  try {
+    const intergerMonzo: number[] = equave.monzo.toIntegerMonzo();
+    return intergerMonzo.reduce(
+      (
+        primeIndex: number | null,
+        component: number,
+        i: number
+      ): number | null => {
+        if (component === 1 && !primeIndex) {
+          return i;
+        }
+        if (component !== 0) {
+          throw new Error("Equave is not a prime number");
+        }
+        if (component === 0) {
+          return primeIndex;
+        }
+        return null;
+      },
+      null
+    );
+  } catch {
+    return null;
+  }
+}
+
+const nodes: ComputedRef<Node[] | string> = computed(() => {
+  try {
+    const equavePrimeIndex = maybeGetPrimeEquave(props.scale.equave);
+    if (equavePrimeIndex === null) {
+      return "Equave must be a prime number";
+    }
+    if (props.scale.intervals.length <= 1) {
+      return "A scale must contain at least two intervals";
+    }
+    return props.scale.intervals.map(({ monzo, name: label }, index) => {
+      const node: Node = monzo.toIntegerMonzo().reduce(
+        (node, component, i): Node => {
+          const { x, y } = node;
+          if (i === equavePrimeIndex) {
+            return node;
+          }
+
+          const componentVal = component.valueOf();
+          if (componentVal === 0) {
+            return node;
+          } else {
+            const vector = coords[i] || { x: 0, y: 0 };
+            return {
+              ...node,
+              x: x + vector.x * componentVal,
+              y: y + vector.y * componentVal,
+            };
+          }
+        },
+        { x: 0, y: 0, label, index, fill: "white" }
+      );
+      return node;
+    });
+  } catch {
+    return "Cannot make lattice of non JI scale.";
+  }
+});
+
+const nodesByLabel: ComputedRef<{ [key: string]: Node }> = computed(() =>
+  Array.isArray(nodes.value)
+    ? nodes.value.reduce((acc, node) => ({ ...acc, [node.label]: node }), {})
+    : {}
+);
+
+const edges = computed(() => {
+  try {
+    if (!Array.isArray(nodes.value)) {
+      return [];
+    }
+    const combinations = getPairCombinations(props.scale.intervals);
+    return combinations
+      .map(([a, b]) => {
+        return {
+          distance: monzoEuclideanDistance(
+            a.monzo.toIntegerMonzo().slice(1),
```

Comment:
Was the lattice supposed to work with all prime equaves? This line assumes an octave.